src/deploy/functions/containerCleaner.ts

```diff
@@ -39,6 +40,25 @@ const SUBDOMAIN_MAPPING: Record<string, string> = {
   "australia-southeast1": "asia",
 };
 
+async function retry<Return>(func: () => Promise<Return>): Promise<Return> {
+  const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+  const MAX_RETRIES = 3;
+  const INITIAL_BACKOFF = 100;
+  let retry = 0;
+  while (true) {
+    try {
+      return await func();
+    } catch (error) {
+      logger.debug("Failed docker command with error", error);
+      retry += 1;
+      if (retry >= MAX_RETRIES) {
+        throw new FirebaseError("Failed to clean up artifacts", { original: error });
+      }
+      await sleep(Math.pow(INITIAL_BACKOFF, retry - 1));
```

Comment:
```suggestion
      await sleep(Math.pow(INITIAL_BACKOFF, retry));
```

Otherwise, it will sleep for 1ms on first iteration I think.

Reply:
It's intentional that the first retry does not have a backoff. Not all failures need congestion control (and if this library really was about congestion control I'd add jitter)